src/providers/java_maven.js

```diff
@@ -72,36 +72,66 @@ function dotGraphToPurl(root) {
 	let name = parts[1]
 	let version = parts[3].replaceAll("\"","")
 	return new PackageURL('maven',group,name,version,undefined,undefined);
-
-
 }
 
 /**
  *
- * @param {String} dotGraphList Dot Graph tree String of the pom.xml manifest
+ * @param {String} dotGraphList Text graph String of the pom.xml manifest
  * @param {[String]} ignoredDeps List of ignored dependencies to be omitted from sbom
  * @return {String} formatted sbom Json String with all dependencies
  * @private
  */
-function createSbomFileFromDotGraphFormat(dotGraphList, ignoredDeps) {
+function createSbomFileFromTextFormat(dotGraphList, ignoredDeps) {
+	let lines = dotGraphList.split(EOL);
 	// get root component
-	let lines = dotGraphList.replaceAll(";","").split(EOL);
-	let root = lines[0].split("\"")[1];
+	let root = lines[0];
 	let rootPurl = dotGraphToPurl(root);
-	lines.splice(0,1);
-	let sbom = new Sbom()
-	sbom.addRoot(rootPurl)
-	lines.forEach(pair => {
-		if(pair.trim() !== "}") {
-			let thePair = pair.split("->")
-			if(thePair.length === 2) {
-				let from = dotGraphToPurl(thePair[0].trim())
-				let to = dotGraphToPurl(thePair[1].trim())
-				sbom.addDependency(sbom.purlToComponent(from), to)
-			}
+	let sbom = new Sbom();
+	sbom.addRoot(rootPurl);
+	calculateTree(root, 0, lines.slice(1), sbom);
+	return sbom.filterIgnoredDepsIncludingVersion(ignoredDeps).getAsJsonString();
+}
+
+// const DEP_REGEX = "(?:([-a-zA-Z0-9.]+:){4}[-a-zA-Z0-9.]+)"
+const DEP_REGEX = /\s*(|s{2})*[+\][-]\s[(]?(.*):(compile|runtime|provided|test|system).*/;
+
+function calculateTree(src, srcDepth, lines, sbom) {
+	if(lines.length === 0) {
+		return
+	}
+	if((lines.length === 1 && lines[0].trim() === ""))
+	{
+		return
+	}
+	let index = 0;
+	let target = lines[index];
+	let targetDepth = getDepth(target);
+	while(targetDepth > srcDepth && index < lines.length) {
+		if(targetDepth == srcDepth + 1) {
```

Comment:
Although it works ( because it's a primitive type) , i think we better change it to tripe =,  and we'll remove this linting warning as a bonus